samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -345,86 +347,34 @@ private     RenderedTemplate    getRenderedTemplate             (String template
 
         return headers;
     }
-
-    // loadTemplate from ValidationLogicImpl.java
-    /**
-     * Load and register one or more email templates (contained in the given
-     * .xml file) with the email template service
-     *
-     * @param templateFileName - the name of the .xml file to load
-     * @param templateRegistrationString - the key (name) of the template to be saved to the service
-     */
-    @SuppressWarnings("unchecked")
-    private     void                loadTemplate                    (String templateFileName, String templateRegistrationString) {
-        LOG.info(this + " loading template " + templateFileName);
-
-        SecurityAdvisor yesMan = new SecurityAdvisor() {
-            public SecurityAdvice isAllowed( String userId, String function, String reference ) {
-                return SecurityAdvice.ALLOWED;
-            }
-        };
-
-        try {
-            // Push the yesMan SA on the stack and perform the necessary actions
-            securityService.pushAdvisor(yesMan);
-
-            // Load up the resource as an input stream
-            InputStream input = SamigoETSProviderImpl.class.getClassLoader().getResourceAsStream("" + templateFileName);
-            if ( input == null ) {
-                LOG.error( "Could not load resource from '" + templateFileName + "'. Skipping ..." );
-            } else {
-                // Parse the XML, get all the child templates
-                Document document = new SAXBuilder().build( input );
-                List<?> childTemplates = document.getRootElement().getChildren( "emailTemplate" );
-                Iterator<?> iter = childTemplates.iterator();
-
-                // Create and register a template with the service for each one found in the XML file
-                while ( iter.hasNext() ) {
-                    xmlToTemplate( (Element) iter.next(), templateRegistrationString );
-                }
-            }
-        } catch (JDOMException | IOException e) {
-            LOG.error("loadTemplate error: ", e);
-        } finally { // Pop the yesMan SA off the stack (remove elevated permissions)
-            securityService.popAdvisor(yesMan);
+	
+    @Override
+    public void notifyAutoSubmitFailures(int count)
+    {
+        boolean notifyOn = serverConfigurationService.getBoolean(SamigoConstants.SAK_PROP_AUTO_SUBMIT_ERROR_NOTIFICATION_ENABLED, false);
+        if (!notifyOn)
+        {
+            return;
         }
-    }
+        String supportAddress = StringUtils.defaultIfBlank(serverConfigurationService.getString(SamigoConstants.SAK_PROP_SUPPORT_EMAIL_ADDRESS), fromAddress);
+        String toAddress = StringUtils.defaultIfBlank(serverConfigurationService.getString(SamigoConstants.SAK_PROP_AUTO_SUBMIT_ERROR_NOTIFICATION_TO_ADDRESS), supportAddress);
```

Comment:
I don't like the new pattern either, but here is the javadoc from SCS:

/**
	 * Access some named configuration value as a string.
	 * 1) IF "name=value" THEN this will return "value"
	 * 2) IF "name=" THEN this will return null
	 * 3) IF name is not defined in the config THEN this will return the provided default value
	 * 
	 * @param name The configuration value name (or key).
	 * @param dflt The value to return if not found in the config.
	 * @return The configuration value for this name OR null if defined as 'blank' OR default value if not defined.
	 */

In BaseConfigurationService.java there are two places where it can call StringUtils.trimToNull() and return the result. So, I couldn't trust that the default value would always be applied, and I needed the code running locally so this is what I went with. Certainly open to changing it.